src/write.rs

```diff
@@ -117,7 +123,13 @@ where
         // destructors of any of the values that are in our map, as they'll all be called when the
         // last read handle goes out of scope.
         for (_, mut vs) in self.w_handle.as_mut().unwrap().data.drain() {
-            for v in vs.drain(..) {
+            #[cfg(not(feature = "smallvec"))]
+            let drain = vs.drain(..);
+
+            #[cfg(feature = "smallvec")]
+            let drain = vs.drain();
```

Comment:
I wonder: to avoid potentially shooting ourselves in the leg later, would it work to do:
```rust
let drain;

#[cfg(not(feature = "smallvec"))]
drain = vs.drain(..);

#[cfg(feature = "smallvec")]
drain = vs.drain();
```
That way we can never accidentally call `.drain()` twice, because it'd be a double-assignment to a non-mutable variable. I _think_ that should compile, but if not, leave it as-is.

Reply:
The two variations are mutually exclusive, so it doesn't really matter. If they were to somehow both exist, that would be a pretty major bug in the compiler.

Reply:
I agree, but writing it as above ensures that we don't accidentally write `cfg`s later that are _not_ mutually exclusive :)

Reply:
Well, upon trying it out, it seems we can't even do it that way.

```rust
let drain;

#[cfg(...)]
drain = whatever;
```

counts as an expression, and `#[cfg(...)]` is not yet supported for expressions, only declarations.

Reply:
Bah, that's annoying. Oh well, keep it the way it is then :)